### PR TITLE
Fixing Jace, Architect of Thought with concurrent search library (for issue #5357)

### PR DESCRIPTION
**Pull request by mooretc2**
This is what I have so far, currently replacing a rules discrepancy with a bug from #5357.

### CODE REVIEW

**Comment by Zzooouhh** [comment]
Couldn't the concurrent search be avoided if the ability simply dumped all the players in range into a list that you can search libraries for in any order? Removing a player from the list only if you find and exile a card.

**Comment by mooretc2** [comment]
Thank you for the thoughts. I assumed as much about the concurrent card code breaking the game before I got into it, but I figured it was worth a shot before adding more complexity. Looking at extending the player/target logic was one of the things I thought might lead to a solution, so I'll start looking into that.

File: Mage.Tests/src/test/java/org/mage/test/player/TestPlayer.java [hunk 1]
```diff
@@ -2072,6 +2072,10 @@ public boolean searchLibrary(TargetCardInLibrary target, Game game, UUID targetP
         return computerPlayer.searchLibrary(target, game, targetPlayerId, triggerEvents);
     }
 
+    public void lookAtAllLibraries(Ability source, Game game) {
```
Comment (JayDi85) [inline Mage.Tests/src/test/java/org/mage/test/player/TestPlayer.java]: @Override was missing

Comment (mooretc2) [reply Mage.Tests/src/test/java/org/mage/test/player/TestPlayer.java]: Fixed in latest commit

**Pull request merged**